### ISSUE TITLE
download post-decompress write path writes out in 4kiB blocks

### ISSUE THREAD

**Issue by mharvey-jt** [cvmfs/cvmfs#3200]
Looking at the performance of the downloader, I see that the decompressor is working in 4k blocks, even when it should be able to make a single larger write. (Observed with strace).
This seems to be due to writing through `Transaction.buffer`,  https://github.com/cvmfs/cvmfs/blob/606f20e3a9c54861b06180725d4a4b080aacce16/cvmfs/cache_posix.h#L125

Since a sqlite db file seems never to be < 16kiB, perhaps it makes sense to increase this size somewhat?

**Comment by jblomer** [comment]
I agree in principle.  It requires a little bit of thinking.  We should use the information about the size of the object because many files in the cache are smaller than 4kB, so not always we'd like a larger buffer.  Also, we allocate the `Transaction` object on the stack, which limits its (direct) size.  Perhaps we can optionally heap allocate a large buffer.
